Repo: yarn build doesn't build website-eslint

### Suggestion

Setting up a new repository locally in theory requires just `yarn` as pre-work before `yarn start`. In theory `yarn` should both install and run a full build (it calls to `yarn build`). But `yarn build` specifically excludes the `website-eslint` package:

https://github.com/typescript-eslint/typescript-eslint/blob/18f8d74cb9ba8b72fb55aa40872fe8bb18549f92/package.json#L26

That means if you `yarn start` after just `yarn`, the website gives a complaint like:

```shell
[ERROR] Error: Cannot find module '/Users/josh/repos/typescript-eslint-upstream/node_modules/@typescript-eslint/website-eslint/dist/index.js'
```

Is there a reason `yarn build` excludes `website-eslint`?

@Zamiell's #7395 touched on this too.

--- a/packages/website/package.json
+++ b/packages/website/package.json
@@ -11,7 +11,7 @@
     "stylelint:fix": "stylelint \"src/**/*.css\" --fix",
     "lint": "nx lint",
     "serve": "docusaurus serve",
-    "start": "docusaurus start",
+    "start": "nx start",
     "swizzle": "docusaurus swizzle",
     "test": "playwright test",
     "typecheck": "tsc -b ./tsconfig.json"

--- a/packages/website/project.json
+++ b/packages/website/project.json
@@ -4,6 +4,14 @@
   "type": "library",
   "implicitDependencies": [],
   "targets": {
+    "start": {
+      "dependsOn": ["website-eslint:build"],
+      "executor": "nx:run-commands",
+      "options": {
+        "cwd": "packages/website",
+        "command": "docusaurus start"
+      }
+    },
     "lint": {
       "executor": "@nx/linter:eslint",
       "outputs": ["{options.outputFile}"],

